From bcc75b259a53e2f438e294162abb05553dc8b42d Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Thu, 04 Jul 2024 14:23:54 +0300
Subject: [PATCH] [FSW-2704] fs1: temp spi dma fix for auto mode

with this patch, removed new spi dma enhancement
bits of mask_crc and restart_crc. once we have
proper rtl implementation for new spi dma, will
revert this.

Change-Id: Ib96c909139ea8c4ec65d0fc6d87c6529c90d315e
---

diff --git a/arch/fs1/arc/dev_dma.c b/arch/fs1/arc/dev_dma.c
index 805446e..83b0ddf 100644
--- a/arch/fs1/arc/dev_dma.c
+++ b/arch/fs1/arc/dev_dma.c
@@ -19,8 +19,8 @@
 #define DMA_DST_BLOCK_SIZE	4
 
 void dma_copy_image_internal(uint16_t image_size, uint32_t flash_offset,
-		uint64_t dest_offset, uint8_t fw_sram_dst, uint8_t restart_crc,
-		uint8_t mask_crc)
+		uint64_t dest_offset, uint8_t fw_sram_dst, __unused uint8_t restart_crc,
+		__unused uint8_t mask_crc)
 {
 	uint32_t reg_spi_dma_ctrl_s =  0;
 	uint32_t spi_dma_dst_addr_l = DMA_ADDRESS_LOW(dest_offset);
@@ -40,10 +40,10 @@
 	reg_spi_dma_ctrl_s |= fw_sram_dst << PARC_GLOBAL_CONF_SPI_DMA_CTRL_DST_SRAM_S;
 
 	/* 0: continue crc calc 1: restart crc*/
-	reg_spi_dma_ctrl_s |= restart_crc << PARC_GLOBAL_CONF_SPI_DMA_CTRL_CRC_RESTART_S;
+	// reg_spi_dma_ctrl_s |= restart_crc << PARC_GLOBAL_CONF_SPI_DMA_CTRL_CRC_RESTART_S;
 
 	/* 0: pass crc error to status 1: mask crc error */
-	reg_spi_dma_ctrl_s |= mask_crc << PARC_GLOBAL_CONF_SPI_DMA_CTRL_MASK_CRC_ERR_S;
+	// reg_spi_dma_ctrl_s |= mask_crc << PARC_GLOBAL_CONF_SPI_DMA_CTRL_MASK_CRC_ERR_S;
 
 	/* # of 4bytes to DMA copy */
 	reg_spi_dma_ctrl_s |= image_size_4b << PARC_GLOBAL_CONF_SPI_DMA_CTRL_MEM_SIZE_S;
